lumigator/python/mzai/backend/backend/api/routes/jobs.py

```diff
@@ -103,20 +105,57 @@ def list_jobs(
     )
 
 
-@router.get("/{job_id}")
-def get_job(service: JobServiceDep, job_id: UUID) -> Job:
+@router.get("/{job_spec}")
+def get_job(
+    service: JobServiceDep,
+    job_spec: str,
+    skip: int = 0,
+    limit: int = 100,
+) -> Job | ListingResponse[Job]:
     """Attempts to retrieve merged job data from the Lumigator repository and Ray.
+    If a valid UUID is provided, a single record is returned. If a valid
+    job type (currently `inference` or `evaluation`) is provided, then a list
+    of all jobs of the appropriate type is returned. Any other input results
+    in an error.
 
     NOTE: Lumigator repository data takes precedence over Ray metadata.
     """
-    job = service.get_job(job_id)
-    ray_job = _get_ray_job(job_id)
-
-    # Combine both types of response.
-    x = ray_job.model_dump()  # JobSubmissionResponse
-    y = job.model_dump()  # JobResponse
-    merged = {**x, **y}
-    return Job(**merged)
+    try:
+        job_id = UUID(job_spec)
+        job = service.get_job(job_id)
+        ray_job = _get_ray_job(job_id)
+
+        # Combine both types of response.
+        x = ray_job.model_dump()  # JobSubmissionResponse
```

Comment:
if we're looking to understand the type here, we can annotate it: 

```python
x: JobSubmissionResponse = ray_job.model_dump()
```

Reply:
Hmmm, I'm not sure I wrote that, but type annotations are fine for me. IIRC this would need to be taken from ray (but we do have those dependencies and can import the types, right?).

Reply:
> I'm not sure I wrote that

I mean I just copied the existing approach of merging the Ray info together with our own.